src/config/nfs.md

```diff
@@ -0,0 +1,142 @@
+```
+
+Check out [nfs(5)](https://man.voidlinux.org/nfs.5) for more information about
+the available options.
+
+Due to [void-runit#16](https://github.com/void-linux/void-runit/issues/16), if
+you mount an nfs volume, you will have to do some manual maintenance to avoid
+your system refusing to shut down.
+
+As root, create a shell script `/etc/rc.pre-shutdown`:
+
+```
+echo "   trying to unmount network shares..."
+echo "   https://github.com/void-linux/void-runit/issues/16"
+
+umount -a -r -t nfs,nfs4,cifs
+
+if grep -qs -e 'nfs ' -e 'nfs4 ' -e 'cifs ' /proc/mounts; then
+   echo
+   echo "...... failed to unmount network some network shares ......"
+   rem_shares=$(grep -e 'nfs ' -e 'nfs4 ' -e 'cifs ' /proc/mounts)
+   if echo $rem_shares | grep -qs 'rw,'; then
+         echo "failed to mount read only as well"
+         echo "will attempt forced, lazy unmount"
+         echo "if this hangs then will need to hard shutdown..."
+         umount -a -f -l -t nfs,nfs4,cifs
+   elif echo $rem_shares | grep -qs 'ro,'; then
+         echo "shares were mounted read only instead."
+         echo "hopefully this is ok. If hangs...hard shutdown required"
+   else
+         echo "I should not be here..."
+   fi
+   echo
+fi
+```
+
+Make that script executable with `chmod a+x /etc/rc.pre-shutdown`, then edit
+`/etc/runit/3` to insert a call to it after `chmod 100 reboot` and before
+`Waiting for services to stop`. Edit the top of `/etc/runit/3` so it looks
+something like this:
+
+```
+#!/bin/sh
+# vim: set ts=4 sw=4 et:
+
+PATH=/usr/bin:/usr/sbin
+
+. /etc/runit/functions
+detect_virt
+[ -r /etc/rc.conf ] && . /etc/rc.conf
+
+if [ -e /run/runit/reboot ]; then
+    chmod 100 /run/runit/reboot
+fi
+
+echo
+msg "Starting pre-shutdown hook https://github.com/void-linux/void-runit/issues/16"
+[ -x /etc/rc.pre-shutdown ] && /etc/rc.pre-shutdown
+
+msg "Waiting for services to stop..."
+sv force-stop /var/service/*
+sv exit /var/service/*
+
+[ -x /etc/rc.shutdown ] && /etc/rc.shutdown
+```
+
+Once this is complete, your machine should be able to reboot.
+
+## Setting up a server (NFSv4, kerberos disabled)
+
+If you are interested in the server, the first step is the same:
+
+```
+# xbps-install -S nfs-utils
+```
+
+Edit the `/etc/exports` file:
+
+```
+# /etc/exports
+#
+# See exports(5) for more information.
+#
+# Use exportfs -r to reread
+# /export       192.168.1.10(rw,no_root_squash)
+/storage/foo    *.local(rw,no_subtree_check,no_root_squash)
+```
+
+With this file we are exporting to any host in the local domain the /storage/foo
+directory as read/write. For information about what the other two options do,
+refer to exports(5).
+
+Start and enable the required services:
+
+```
+# ln -s /etc/sv/rpcbind /var/service
+# ln -s /etc/sv/statd /var/service
+# ln -s /etc/sv/nfs-server /var/service
+```
+
+The shares should now be working. If that's not the case please check
+/var/log/messages (if using a syslog daemon). Use the `showmount(8)` utility to
```

Comment:
Actually, make that:

"If not, and you are using a syslog daemon, check \`/var/log/messages\` for possible information about the cause of the error. Use the \[showmount(8)\](https://man.voidlinux.org/showmount.8) utility to"